### PR TITLE
CB-14897 Add client configuration with retry to AWSSecurityTokenService everyw…

### PR DESCRIPTION
**Pull request by topolyai5**
…here. Unfortunately, the retry and tracing are missing during the AWS assume. This functionality added with this PR

### CODE REVIEW

**Comment by keyki** [comment]
Missing CB ticket + add a detailed commit message please.

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsClient.java [hunk 1]
```diff
@@ -323,8 +323,4 @@ private AwsSessionCredentialProvider createAwsSessionCredentialProvider(AwsCrede
         proxyFactory.addAspect(new AmazonClientExceptionHandler(awsCredentialView, region, sdkClientExceptionMapper));
         return proxyFactory.getProxy();
     }
-
-    public void setAwsSessionCredentialClient(AwsSessionCredentialClient awsSessionCredentialClient) {
```
Comment (Bajzathd) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsClient.java:327]: If I remember correctly it was there to enable setter bean injection, did something change that it is no longer required?

**Comment by lajosrodek** [comment]
I think this is supposedly for [CB-14897](https://jira.cloudera.com/browse/CB-14897).

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsClient.java (hunk 1 above)
Comment (topolyai5) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsClient.java:327]: @akanto removed the bean injection in this PR: https://github.com/hortonworks/cloudbreak/commit/2436b25fd0abd36a201cf92e35e73513f658ffc9

**Review by Bajzathd — APPROVED** [review]

**Review by oleewere — APPROVED** [review]
lgtm

**Pull request merged**